src/api/app/controllers/webui/project_controller.rb

```diff
@@ -707,21 +708,29 @@ def save_meta
   end
 
   def prjconf
-    begin
-      @config = @project.source_file('_config')
-    rescue ActiveXML::Transport::NotFoundError
-      flash[:error] = "Project _config not found: #{params[:project]}"
-      redirect_to :controller => 'project', :nextstatus => 404 and return
+    @content = @project.config.to_s
+    unless @content
+      flash[:error] =  @project.config.errors.full_messages.to_sentence
+      redirect_to controller: 'project', nextstatus: 404
+      return
     end
   end
 
   def save_prjconf
     authorize @project, :update?
 
-    check_ajax
-    frontend.put_file(params[:config], :project => params[:project], :filename => '_config')
-    flash[:notice] = 'Project Config successfully saved'
-    render text: 'Config successfully saved', content_type: 'text/plain'
+    params[:user] = User.current.login
+    query = params.slice(:user, :comment)
+
+    content = @project.config.save(query, params[:config])
+
+    if content
+      flash.now[:success] = 'Config successfully saved!'
+      render layout: false, partial: 'layouts/webui/flash', object: flash
```

Comment:
Yes! What's wrong with it?
Before we returned no feedback what happened to the user at all! Only the save button stayed red if something went wrong ...